docs/kubernetes/routegroups.md

```diff
@@ -0,0 +1,628 @@
+
+Skipper is an extensible HTTP router with rich route matching, and request flow and traffic shaping
+capabilities. Through its integration with Kubernetes, it can be used in the role of an ingress controller for
+forwarding incoming external requests to the right services in a cluster. Kubernetes provides the Ingress
+specification to define the rules by which an ingress controller should handle the incoming traffic. The
+specification is simple and generic, but doesn't offer a straightforward way to benefit from Skipper's rich HTTP
+related functionality.
+
+## RouteGroups
+
+A RouteGroup is a custom Kubernetes resource definition. It provides a way to define the ingress routing for
+Kubernetes services. It allows route matching based on any HTTP request attributes, and provides a clean way for
+the request flow augmentation and traffic shaping. It supports higher level features like gradual traffic
+switching, A/B testing, and more.
+
+Example:
+
+```yaml
+apiVersion: zalando.org/v1
+kind: RouteGroup
+metadata:
+  name: my-routes
+spec:
+  backends:
+  - name: variant-a
+    type: service
+    serviceName: service-a
+    servicePort: 80
+  - name: variant-b
+    type: service
+    serviceName: service-b
+    servicePort: 80
+  defaultBackends:
+  - backendName: variant-b
+  routes:
+  - pathSubtree: /
+    filters:
+    - responseCookie("canary", "A")
+    predicates:
+    - Traffic(.1)
+    backends:
+    - backendName: variant-a
+  - pathSubtree: /
+    filters:
+    - responseCookie("canary", "B")
+  - pathSubtree: /
+    predicates:
+    - Cookie("canary", "A")
+    backends:
+    - backendName: variant-a
+  - pathSubtree: /
+    predicates:
+    - Cookie("canary", "B")
+```
+
+(See a more detailed explanation of the above example [further down](#gradual-traffic-switching) in this
+document.)
+
+Links:
+
+- [RouteGroup semantics](../routegroup-crd/)
+- [CRD definition](https://github.com/zalando/skipper/blob/master/dataclients/kubernetes/deploy/apply/routegroups_crd.yaml)
+
+## Current Limitations
+
+As mentioned above, the interagtion with load balancer and DNS configuration solutions is a work in progress.
+This means that when using route groups, updating the DNS entries and the load balancer configurations based on
+the host names defined in a route group, currently needs to be done manually. As near milestones, the
+integration will be done in:
+
+- [Extenral DNS](https://github.com/kubernetes-sigs/external-dns)
+- [Kubernetes Ingress Controller for AWS](https://github.com/zalando-incubator/kube-ingress-aws-controller)
+
+## Installation
+
+The definition file of the CRD can be found as part of Skipper's source code, at:
+
+https://github.com/zalando/skipper/blob/master/dataclients/kubernetes/deploy/apply/routegroups_crd.yaml
+
+To install it manually in a cluster, assuming the current directory is the root of Skipper's source, call this
+command:
+
+```
+kubectl apply -f dataclients/kubernetes/deploy/apply/routegroups_crd.yaml
```

Comment:
% kubectl apply -f https://raw.githubusercontent.com/zalando/skipper/feature/routegroup-crd/dataclients/kubernetes/deploy/apply/routegroups_crd.yaml
customresourcedefinition.apiextensions.k8s.io/routegroups.zalando.org created

Reply:
good to know that this is possible with apply -f, but i would not promote it in the docs. Note that we may new releases, while some want to use the edge from master, etc.